osh/glob_.py

```diff
@@ -96,6 +91,23 @@ def GlobEscape(s):
     return pyutil.BackslashEscape(s, GLOB_META_CHARS)
 
 
+def GlobEscapeUnquotedSubstitution(s):
+    # type: (str) -> str
+    """For the substitution results such as $v with v='a\*b.txt'."""
+
+    # When glob expansion is not performed, unquoted substitutions need to be
+    # literally preserved in the final result.  A backslash inside unquoted
+    # substitutions should be treated as escaping in glob expansion, yet it
+    # needs to be preserved in the final result without glob expansion.  This
+    # means that a backslash inside unquoted substitutions is treated
+    # differently from quoted backslash, so we need a distinct representaton.
+    # We choose \@ here.
+    #
+    # XXX--This representation is affected by the known IFS='\' bug, but the
```

Comment:
OK I see the problem here ... But it seems like we probably should not use `\@` in the long term ??

I guess that's what this comment is indicating

Do we know what kind of representation we will want, or maybe we need to look at IFS=\ more  before figuring it out?

Reply:
I already fixed the `IFS=\` bug in my local branch based on this glob fix.

Also, I considered different representations and different approaches, but I think this is the most efficient and simple way.

Reply:
Oh cool, is it possible to send a PR for that before this is merged?  I guess you are saying they are not strictly related, but I am curious about it

Reply:
I submitted Draft PR #2353 (but this includes also all the changes from #2349 and #2350). The substantial change of #2353 can be checked from [`5abab65..bs-IFS`](https://github.com/oils-for-unix/oils/compare/5abab656cd94b3f12001834edb5751af48dd7337..bs-IFS).